test/integration/provider/admin/applications/applications_controller_test.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Provider::Admin::ApplicationsTest < ActionDispatch::IntegrationTest
+  class MasterLoggedInTest < Provider::Admin::ApplicationsTest
+    attr_reader :service
+
+    setup do
+      login! master_account
+      @service = master_account.default_service
+      FactoryBot.create(:cinstance, service: service)
+    end
+
+    test 'index retrieves all master\'s provided cinstances except those whose buyer is master' do
+      get provider_admin_applications_path
+
+      assert_response :ok
+      expected_cinstances_ids = master_account.provided_cinstances.not_bought_by(master_account).pluck(:id)
+      assert_same_elements expected_cinstances_ids, assigns(:cinstances).map(&:id)
+    end
+
+    test 'show is visible for all master\'s provided cinstances except those whose buyer is master' do
+      buyer_master, provider_master = service.cinstances.partition { |cinstance| cinstance.user_account_id == Account.master.id }
+      buyer_master.each do |buyer_cinstance|
+        get provider_admin_application_path(buyer_cinstance)
+        assert_response :not_found
+      end
+      provider_master.each do |provider_cinstance|
+        get provider_admin_application_path(provider_cinstance)
+        assert_response :ok
+      end
+    end
+  end
+
+  class ProviderLoggedInTest < Provider::Admin::ApplicationsTest
+    setup do
+      @provider = FactoryBot.create(:provider_account)
+      login! @provider
+    end
+
+    attr_reader :provider
+
+    class Index < ProviderLoggedInTest
+      setup do
+        @service = provider.services.first!
+        plans = FactoryBot.create_list(:application_plan, 2, service: @service)
+        buyers = FactoryBot.create_list(:buyer_account, 2, provider_account: provider)
+        plans.each_with_index { |plan, index| buyers[index].buy! plan }
+      end
+
+      attr_reader :service
+
+      test 'index shows the services column when the provider is multiservice' do
+        provider.services.create!(name: '2nd-service')
+        assert provider.reload.multiservice?
+        get provider_admin_applications_path
+        page = Nokogiri::HTML::Document.parse(response.body)
+        assert page.xpath("//tr").text.match /Service/
```

Comment:
Maybe better to reference which table:

`page.xpath('//table#id/tr/td[text() = 'Service']')`